src/jit/morph.cpp

```diff
@@ -14057,6 +14057,12 @@ GenTree* Compiler::fgMorphModToSubMulDiv(GenTreeOp* tree)
         denominator = fgMakeMultiUse(&tree->gtOp2);
     }
 
+    if (lvaLocalVarRefCounted)
```

Comment:
I don't understand why we can unconditionally call this for either the numerator or the denominator (though for the latter case it is probably not hit in testing because it is only called with a constant denominator except for arm64, at least according to the header comment). Why doesn't this risk an extra increment if it is already a lclVar (i.e. no temp is created)? Another odd thing is that we never seem to use `numerator` and `denominator` in the case where they are non-leaf-nodes, since their original use has been replaced by the comma with the temp assignment on the lhs, and the returned value is (as yet) unused. I think that we should only be cloning these in the leaf case.

---
In reply to: [94655104](https://github.com/dotnet/coreclr/pull/8804#discussion_r94655104) [](ancestors = 94655104)